web/src/composebox_typeahead.js

```diff
@@ -222,18 +223,30 @@ function handle_keydown(e) {
                     // could result in focus being moved to the "Send
                     // button" after sending the message, preventing
                     // typing a next message!
-                    $("#compose-send-button").trigger("focus");
+                    if (popover_menus.is_time_selected_for_schedule()) {
+                        $("#schedule-confirm").trigger("focus");
+                    } else {
+                        $("#compose-send-button").trigger("focus");
+                    }
+
                     e.preventDefault();
                     e.stopPropagation();
                 }
             } else {
                 // Enter
                 if (should_enter_send(e)) {
                     e.preventDefault();
-                    if (
-                        compose_validate.validate_message_length() &&
-                        !$("#compose-send-button").prop("disabled")
-                    ) {
+
+                    if (!compose_validate.validate_message_length()) {
+                        return;
+                    }
+
+                    if (popover_menus.is_time_selected_for_schedule()) {
+                        popover_menus.schedule_message_to_custom_date();
+                        return;
+                    }
+
+                    if (!$("#compose-send-button").prop("disabled")) {
                         compose.finish();
```

Comment:
Like with my previous comment in this file, I think ideally we'd have an abstraction to access the button element (maybe something like a `compose.get_submit_button` helper would be a good idea?) and then we should check whether that button is disabled a few lines up -- it seems likely that conditions that block sending a message should also block scheduling one, and I think this logic circumvents that. 

I'd also be a bit tempted to have the API be that `compose.finish` is a common code path for submitting the compose box regardless of its scheduled-or-not state, but I've not gotten to that part of the implementation, so we can discuss that in the right part of the code.

Reply:
Okay, done.